hibernate-core/src/main/java/org/hibernate/internal/util/QuotingHelper.java

```diff
@@ -70,13 +70,24 @@ public static String unquoteIdentifier(String text) {
 	}
 
 	public static String unquoteStringLiteral(String text) {
+		return unquoteString( text, false );
+	}
+
+	public static String unquoteJavaStringLiteral(String text) {
+		return unquoteString( text, true );
+	}
+
+	public static String unquoteString(String text, boolean unescape) {
 		assert text.length() > 1;
+		int start = 0;
 		final int end = text.length() - 1;
-		final char delimiter = text.charAt( 0 );
+		final char delimiter = unescape && Character.toLowerCase( text.charAt( start ) ) == 'j' ?
+				text.charAt( ++start ) :
+				text.charAt( start );
```

Comment:
The suggestion to pass the start index explicitly is mostly because this conditional expression increment feels like voodoo 😄 

```suggestion
		final char delimiter = text.charAt( start );
```

Reply:
Yep, I guess one shouldn't code late at night 😁 anyways this was a quick WIP, mainly wanted to update the code with the new ideas but still waiting on any news in the discussion.

For now I applied the proposed changes, thanks.

Reply:
Don't worry 😁